backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/security/JwtAuthenticationProvider.kt

```diff
@@ -24,5 +26,22 @@ class JwtAuthenticationProvider : AuthenticationProvider {
         return if (user != null) JwtAuthenticationToken(user, token) else null
     }
 
+    fun extractUsername(authorizationHeader: String): String {
+        return JWT.require(Algorithm.HMAC512(jwtSecuritySecret.toByteArray()))
+                .build()
+                .verify(authorizationHeader.replace("Bearer ", ""))
+                .subject
+    }
+
+    fun getCurrentUser() =
+            try {
+                (SecurityContextHolder.getContext().authentication as JwtAuthenticationToken).userId
+            } catch (exception: ClassCastException) {
+                "user"
```

Comment:
Shouldn't this result in an exception? Under which circumstances is there no token?